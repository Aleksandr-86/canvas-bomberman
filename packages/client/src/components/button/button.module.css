.button {
  padding: 10px 50px;
  border: 5px solid #fff;
  border-radius: 15px;
  background: transparent;
  color: #fff;
  font: inherit;
  font-size: 32px;
  line-height: 1.2em;
  cursor: pointer;
}
