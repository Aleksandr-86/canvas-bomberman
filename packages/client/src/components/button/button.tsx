import classes from './button.module.css'


interface Props {
    title: string
    class?: string
    type?: string
}

const Button = (props: Props) => {
    const { title = 'Кнопка' } = props

    if (props.class) {
        return (
            <button className={`${classes.forum__btn} ${props.class}`}>{title}</button>
        )
    } else {
        return <button className={classes.forum__btn}>{title}</button>
    }
}

export { Button }