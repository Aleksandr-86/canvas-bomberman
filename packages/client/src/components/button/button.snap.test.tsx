import * as renderer from 'react-test-renderer'
import { Button } from './button'

const mockProps = {
  children: `mockChildren`,
  className: `mockClass`,
  onClick: () => undefined,
}

describe(`<Button/> snapshot test cases`, () => {
  test(`<Button/> should render <button className=button> with text`, () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Button/> should render <button className=button mockClass> with mockProps`, () => {
    const tree = renderer
      .create(
        <Button
          type="submit"
          className={mockProps.className}
          onClick={mockProps.onClick}>
          {mockProps.children}
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
