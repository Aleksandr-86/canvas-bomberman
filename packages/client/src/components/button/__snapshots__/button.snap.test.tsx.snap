// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> snapshot test cases <Button/> should render <button className=button mockClass> with mockProps 1`] = `
<button
  className="button mockClass"
  onClick={[Function]}
  type="submit"
>
  mockChildren
</button>
`;

exports[`<Button/> snapshot test cases <Button/> should render <button className=button> with text 1`] = `
<button
  className="button"
>
  Кнопка
</button>
`;
